.github/workflows/publish-embedded-packages.yaml

```diff
@@ -11,19 +11,51 @@ on:
   push:
     branches: [livekit]
 
-env:
-  # We perform a dry run for all events except releases.
-  # This is to help make sure that we notice if the packaging process has become
-  # broken ahead of a release.
-  DRY_RUN: ${{ github.event_name != 'release' }}
-  # We should only use the hard coded test value for a dry run
-  VERSION: ${{ github.event_name == 'release' && github.event.release.tag_name || 'v0.0.0-pre.0' }}
-
 jobs:
+  versioning:
+    name: Versioning
+    runs-on: ubuntu-latest
+    outputs:
+      DRY_RUN: ${{ steps.dry_run.outputs.DRY_RUN }}
+      PREFIXED_VERSION: ${{ steps.prefixed_version.outputs.PREFIXED_VERSION }}
+      UNPREFIXED_VERSION: ${{ steps.unprefixed_version.outputs.UNPREFIXED_VERSION }}
+      TAG: ${{ steps.tag.outputs.TAG }}
+    steps:
+      - name: Calculate VERSION
+        # We should only use the hard coded test value for a dry run
+        run: echo "VERSION=${{ github.event_name == 'release' && github.event.release.tag_name || 'v0.0.0-pre.0' }}" >> "$GITHUB_ENV"
+      - id: dry_run
+        name: Set DRY_RUN
+        # We perform a dry run for all events except releases.
+        # This is to help make sure that we notice if the packaging process has become
+        # broken ahead of a release.
+        run: echo "DRY_RUN=${{ github.event_name != 'release' }}" >> "$GITHUB_OUTPUT"
+      - id: prefixed_version
+        name: Set PREFIXED_VERSION
+        run: echo "PREFIXED_VERSION=${VERSION}" >> "$GITHUB_OUTPUT"
+      - id: unprefixed_version
+        name: Set UNPREFIXED_VERSION
+        # This just strips the leading character
+        run: echo "UNPREFIXED_VERSION=${VERSION:1}" >> "$GITHUB_OUTPUT"
+      - id: tag
+        # latest = a proper release
+        # next = a release-candidate
+        # other = anything else
+        name: Set tag
+        run: |
+          if [[ "${VERSION}" =~ ^v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+            echo "TAG=latest" >> "$GITHUB_OUTPUT"
+          elif [[ "${VERSION}" =~ ^v[0-9]+\.[0-9]+\.[0-9]+-rc\.[0-9]+$ ]]; then
+            echo "TAG=next" >> "$GITHUB_OUTPUT"
+          else
+            echo "TAG=other" >> "$GITHUB_OUTPUT"
+          fi
```

Comment:
I don't think this behaves quite right, or at least inconsistently with our other packages.

`next` may be pointing to something older than `latest` - e.g. in js-sdk 

`next` points to `latest` or to an RC newer than `latest`, never to something older than `latest` otherwise you end up with a confusing testing experience of having to manually figure out whether to install `<pkg>@next` or `<pkg>[@latest]`